testsuite/basic/src/test/java/io/smallrye/faulttolerance/bulkhead/BulkheadTest.java

```diff
@@ -50,22 +53,24 @@ public static JavaArchive createTestArchive() {
     @Test
     public void testWaitingQueue(PingService pingService) throws InterruptedException, ExecutionException {
         int loop = QUEUE_SIZE * 2;
-        CountDownLatch startLatch = new CountDownLatch(loop);
+        CountDownLatch startLatch = new CountDownLatch(QUEUE_SIZE);
         CountDownLatch endLatch = new CountDownLatch(1);
         List<Future<String>> futures = new ArrayList<>();
         for (int i = 0; i < loop; i++) {
             futures.add(pingService.ping(startLatch, endLatch));
         }
-        startLatch.await(500, TimeUnit.MILLISECONDS);
+        assertTrue(startLatch.await(1000L, TimeUnit.MILLISECONDS));
```

Comment:
I think we should probably use something like this instead:
```java
startLatch.await(500, TimeUnit.MILLISECONDS);
Thread.sleep(500);
``` 
The reason is that `startLatch.await()` can eventually return before the waiting queue is full.

Reply:
done